SYCL/ESIMD/api/functional/ctors/ctor_broadcast.hpp

```diff
@@ -0,0 +1,196 @@
+//===-- ctor_broadcast.hpp - Functions for tests on simd broadcast constructor
+//      definition. -------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+///
+/// \file
+/// This file provides functions for tests on simd broadcast constructor.
+///
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include "common.hpp"
+
+namespace esimd = sycl::ext::intel::experimental::esimd;
+
+namespace esimd_test::api::functional {
+
+// Descriptor class for the case of calling constructor in initializer context.
+struct initializer {
+  static std::string get_description() { return "initializer"; }
+
+  template <typename SrcT, typename DstT, int NumElems>
+  static void call_simd_ctor(DstT ref_data, SrcT *const out) {
+    const auto simd_by_init = esimd::simd<SrcT, NumElems>(ref_data);
+    simd_by_init.copy_to(out);
+  }
+};
+
+// Descriptor class for the case of calling constructor in variable declaration
+// context.
+struct var_decl {
+  static std::string get_description() { return "variable declaration"; }
+
+  template <typename SrcT, typename DstT, int NumElems>
+  static void call_simd_ctor(DstT ref_data, SrcT *const out) {
+    const esimd::simd<SrcT, NumElems> simd_by_var_decl(ref_data);
```

Comment:
```suggestion
  static void call_simd_ctor(SrcT ref_data, DstT *const out) {
    const esimd::simd<DstT, NumElems> simd_by_var_decl(ref_data);
```
The source and destination naming is a bit confusing over the test code. Is there anything I am missing here?

Reply:
done in this commit: b12bfe332f0da75f23cdd49b29d10318fd4961db